blivet/devices/stratis.py

```diff
@@ -0,0 +1,87 @@
+# devices/stratis.py
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Vojtech Trefny <vtrefny@redhat.com>
+#
+
+import logging
+log = logging.getLogger("blivet")
+
+from .storage import StorageDevice
+from ..static_data import stratis_info
+from ..size import Size
+from ..storage_log import log_method_call
+from .. import devicelibs
+
+
+class StratisPoolDevice(StorageDevice):
+    """ A stratis pool device """
+
+    _type = "stratis_pool"
+    _resizable = False
+    _packages = ["stratisd"]
+    _dev_dir = "/stratis"
+
+    def read_current_size(self):
+        size = Size(0)
+        if self.exists and self.uuid in stratis_info.pools.keys():
+            size = stratis_info.pools[self.uuid].physical_size
+        return size
+
+    def _destroy(self):
+        """ Destroy the device. """
+        log_method_call(self, self.name, status=self.status)
+        devicelibs.stratis.remove_pool(self.uuid)
+
+
+class StratisFilesystemDevice(StorageDevice):
+    """ A stratis pool device """
+
+    _type = "stratis_filesystem"
+    _resizable = False
+    _packages = ["stratisd"]
+    _dev_dir = "/stratis"
+
+    def _get_name(self):
+        """ This device's name. """
+        if self.pool is not None:
+            return "%s/%s" % (self.pool.name, self._name)
```

Comment:
`name` is a unique identifier in blivet so we can't use just the filesystem name here. I decided to use `<pool>/<filesystem>` to make it similar to LVM where we use `<vg>-<lv>`. We could use the DM name which is unique but `name` is often displayed in the installer UI so we'd need to tell Anaconda to make exception for stratis and use `fsname` when displaying stratis devices.

Reply:
I think it makes a lot of sense to use the simpler format that we use for our symlinks as you suggest here as that's what's going to be display to a user at the command line if they run `stratis filesystem list`.